### ISSUE TITLE
Flag --incompatible_disable_starlark_host_transitions will break stardoc in Bazel 7.0

### ISSUE THREAD

**Issue by keertk** [bazelbuild/stardoc#145]
Incompatible flag `--incompatible_disable_starlark_host_transitions` will be enabled by default in the next major release (Bazel 7.0), thus breaking stardoc. Please migrate to fix this and unblock the flip of this flag.

The flag is documented here: [bazelbuild/bazel#17032](https://github.com/bazelbuild/bazel/issues/17032).

Please check the following CI builds for build and test results:

- [MacOS OpenJDK 11](https://buildkite.com/bazel/bazelisk-plus-incompatible-flags/builds/1365#0185154a-df99-44c8-85da-13fbc4e57b75)
- [Ubuntu 18.04 OpenJDK 11](https://buildkite.com/bazel/bazelisk-plus-incompatible-flags/builds/1365#0185154a-df93-40b1-b851-c95d3b0abf08)
- [Ubuntu 20.04 OpenJDK 11](https://buildkite.com/bazel/bazelisk-plus-incompatible-flags/builds/1365#0185154a-df96-4413-a5af-c1a02077d942)
- [Windows OpenJDK 11](https://buildkite.com/bazel/bazelisk-plus-incompatible-flags/builds/1365#0185154a-df9c-4403-bc65-bee100148b84)

Never heard of incompatible flags before? We have [documentation](https://docs.bazel.build/versions/master/backward-compatibility.html) that explains everything.
If you have any questions, please file an issue in https://github.com/bazelbuild/continuous-integration.

**Comment by sgowroji** [comment]
Hi Team, Gentle ping , Can you share any significant update on this issue. Thanks!

**Comment by tetromino** [comment]
We are not using host transitions in Stardoc itself - but it's used by *many* of our deps (skylib, rules_license, rules_jvm_external, rules_cc).

We should check whether those projects have releases which work with --incompatible_disable_starlark_host_transitions (for skylib e.g. we need to update to 1.4.1).

**Comment by tetromino** [comment]
We're blocked by *Bazel itself* using host transitions in third_party/grpc/build_defs.bzl :)

```
ERROR: Traceback (most recent call last):
	File "/private/var/tmp/_bazel_arostovtsev/4db74ff3612d0bdc1259d9a6e5cd2136/external/io_bazel/third_party/grpc/build_defs.bzl", line 64, column 30, in <toplevel>
		"_protoc": attr.label(
Error in label: 'cfg = "host"' is deprecated and should no longer be used. Please use 'cfg = "exec"' instead.
ERROR: /private/var/tmp/_bazel_arostovtsev/4db74ff3612d0bdc1259d9a6e5cd2136/external/io_bazel/src/main/java/com/google/devtools/build/lib/collect/nestedset/BUILD:16:13: error loading package '@io_bazel//src/main/protobuf': initialization of module 'third_party/grpc/build_defs.bzl' failed and referenced by '@io_bazel//src/main/java/com/google/devtools/build/lib/collect/nestedset:nestedset'
```

**Comment by tetromino** [comment]
Steps to fix:

- [x] bump `bazel_skylib` to 1.4.1 or newer - fixed in f8fab820143a75810e6b7d9f764fc7552b82cbdb
- [ ] switch to `cfg = "exec"` in bazel's own `third_party/grpc` rules - waiting for  https://github.com/bazelbuild/bazel/pull/19192
- [ ] bump `rules_jvm_external` to 5.0 or newer (probably 5.2, since that's what `io_bazel` is on at the moment)
